app/services/report_actions/save_pdf_to_inventory_item.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module ReportActions
+  class SavePdfToInventoryItem
+
+    attr_reader :cell_value
+
+    def initialize(user, team, params)
+      @user   = user
+      @team   = team
+      @params = params
+      load_repository_collaborators
+    end
+
+    def call
+      file  = generate_pdf(@params[:html])
+      asset = create_new_asset(file)
+      cell  = fetch_repository_cell
+      cell.destroy if cell
+      @cell_value = create_new_cell_value(asset)
```

Comment:
Yes, you are right but I had spent the most time on the frontend part and then run out of time dedicated to this task to architect better backend code. Also, this service has too many dependencies that will be nice that we extract in the model or even better repository objects and it will also be highly recommended to heavily unit test this service but I do not think it is possible to do all that in 16 hours what is dedicated to this task. Should we open a new task for refactor that? Can I spend an additional day to test/refactor?